test/utils/noderesourcetopologies/noderesourcetopologies.go

```diff
@@ -79,19 +92,29 @@ func GetUpdatedForNode(cli client.Client, ctx context.Context, ref nrtv1alpha2.N
 			klog.Errorf("cannot get the updated NRT object %s/%s", ref.Namespace, ref.Name)
 			return false, err
 		}
-		// very cheap test to rule out false negatives
-		if updatedNrt.ObjectMeta.ResourceVersion == ref.ObjectMeta.ResourceVersion {
-			klog.Warningf("NRT for %q resource version didn't change", ref.Name)
+		equalZones = isEqualNRT(ref, updatedNrt)
+		if equalZones {
 			return false, nil
 		}
-		equalZones = apiequality.Semantic.DeepEqual(ref.Zones, updatedNrt.Zones)
-		klog.Infof("NRT change: updated to %s", e2enrt.ToString(updatedNrt))
-		return !equalZones, nil
+		return true, nil
 	})
 	klog.Infof("NRT change: finished, equalZones=%v", equalZones)
 	return updatedNrt, err
 }
 
+func isEqualNRT(initialNrt, updatedNrt nrtv1alpha2.NodeResourceTopology) bool {
+	// very cheap test to rule out false negatives
+	if updatedNrt.ObjectMeta.ResourceVersion == initialNrt.ObjectMeta.ResourceVersion {
+		klog.Warningf("NRT %q resource version didn't change", initialNrt.Name)
+		return true
+	}
+	equalZones := apiequality.Semantic.DeepEqual(initialNrt.Zones, updatedNrt.Zones)
```

Comment:
I wonder if it takes the same time to update the different resources on NRT. for example if the consumed devices are updated by the same time that the consumed cpus are. 
this PR is definitely needed. I'll have the tests triggered on this once merged to make sure it's stable for all resources checked.

Reply:
it's a good point to explore. I think the code is robust enough atm to deal with that, but this is something we should totally look into for the future

Reply:
What about `AttributeList` https://github.com/k8stopologyawareschedwg/noderesourcetopology-api/blob/master/pkg/apis/topology/v1alpha2/types.go#L54?

Reply:
Good point. This also should be addressed in the followup. The saving grace is the only mutating field (PFP) is eventually a function of the content of the `Zones`.

Reply:
filed https://github.com/openshift-kni/numaresources-operator/issues/609 for tracking